ts/Extensions/Annotations/Controllables/ControllableEllipse.ts

```diff
@@ -0,0 +1,304 @@
+    }
+
+    /**
+     * Set the radius X.
+     *
+     * @param {number} rx a radius in x direction to be set
+     */
+    public setXRadius(rx: number): void {
+        this.options.rx = rx;
+    }
+
+    /**
+     * Set both radius properties.
+     *
+     * @param {number} rx a radius in x direction to be set
+     * @param {number} ry a radius in y direction to be set
+     */
+    public setRadius(rx?: number, ry?: number): void {
+        if (rx) {
+            this.options.rx = rx;
+        }
+        if (ry) {
+            this.options.ry = ry;
+        }
+    }
+    /**
+     * Set the angle of the ellipse.
+     *
+     * @param {number} angle the value of the angle on which the ellipse
+     * is rotated (clockwise driection).
+     */
+    public setAngle(angle: number): void {
+        this.angle = angle;
+    }
+
+    /**
+     * Save the reference point positions, to pin the ellipse to the axes.
+     *
+     * @param x x position of the center of the ellipse
+     * (in pixels or in xAxis value if xAxis is defined)
+     * @param y y position of the center of the ellipse
+     * (in pixels or in yAxis value if yAxis is defined)
+     * @param rx x radius of the ellipse in pixels
+     * @param ry y radius of the ellipse in pixels
+     * @param angle angle in degrees
+     */
+    public savePoints(x?: number, y?: number, rx?: number, ry?: number, angle?: number): void {
+        const xAxis = this.chart.xAxis[(this.options.point as any).xAxis],
+            yAxis = this.chart.yAxis[(this.options.point as any).yAxis],
+            position = this.anchor(this.points[0]).absolutePosition;
+        let points;
+        x = x || position.x;
+        y = y || position.y;
+        rx = rx || this.options.rx;
+        ry = ry || this.options.ry;
+        angle = angle || this.angle;
+        if (xAxis && yAxis) {
+
+            const pointX1 = x - rx * Math.cos((angle * Math.PI) / 180),
+                pointY1 = y - rx * Math.sin((angle * Math.PI) / 180),
+                pointX2 = x + ry * Math.sin((angle * Math.PI) / 180),
+                pointY2 = y - ry * Math.cos((angle * Math.PI) / 180);
+            points = [{
+                x: xAxis.toValue(pointX1),
+                y: yAxis.toValue(pointY1)
+            }, {
+                x: xAxis.toValue(pointX2),
+                y: yAxis.toValue(pointY2)
+            }];
+            this.referencePoints = points;
+        }
+    }
+
+    /**
+     * Retrieve the attributes needed to plot
+     * the ellipse from the reference points.
+     */
+    public getAttrsFromPoints(): EllispseShapeSVGOptions {
+        const points = this.referencePoints,
+            position = this.anchor(this.points[0]).absolutePosition,
+            xAxisIndex = (this.options.point as any).xAxis,
+            yAxisIndex = (this.options.point as any).yAxis,
+            cx = position.x,
+            cy = position.y;
+        if (defined(yAxisIndex) && defined(xAxisIndex)) {
```

Comment:
Is this `if` nececary?

Reply:
yes :)

Reply:
I added a comment to explain what this `if` does